editor/src/templates/editor.tsx

```diff
@@ -450,7 +451,12 @@ export class Editor {
       const reactRouterErrorPreviouslyLogged = hasReactRouterErrorBeenLogged()
 
       const runDomWalker = shouldRunDOMWalker(dispatchedActions, dispatchResult)
-      const shouldRerender = !dispatchResult.nothingChanged || runDomWalker
```

Comment:
I noticed that `shouldRerender` was actually very badly named here, and so changed some of the gating of the code below accordingly. It's worth expanding the diff below to understand what is actually happening now